arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,350 @@
+			};
+		};
+	};
+
+	fragment@4 {
+		target = <&spi5>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 17 GPIO_ACTIVE_LOW>,
+				   <&gpio 18 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi5_pins &spi5_cs_pins>;
+			status = "okay";
+
+			tpm: tpm@0 {
+				compatible = "infineon,slb9670";
+				pinctrl-names = "default";
+				pinctrl-0 = <&tpm_pins>;
+				reg = <0>;
+				spi-max-frequency = <1000000>;	/* <= 43 MHz */
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <22 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+			ksz@1 {
+				compatible = "microchip,ksz8795";
+				reg = <1>;
+				// TODO: interrupt not to be needed? (not used in driver)
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <44 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+
+		};
+	};
+
+	fragment@5 {
+		target = <&spi6>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 6 GPIO_ACTIVE_LOW>,
+				   <&gpio 10 GPIO_ACTIVE_LOW>,
+				   <&gpio 27 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi6_pins &spi6_cs_pins>;
+			status = "okay";
+
+			aout: dac@0 {
+				compatible = "ti,dac7512";
+				reg = <0>;
+				spi-max-frequency = <5000000>;  /* <= 30 MHz */
+				spi-cpol;
+				vcc-supply = <&aout_vref>;
+				status = "okay";
+			};
+
+			ain: adc@1 {
+				compatible = "microchip,mcp3550-50";
+				reg = <1>;
+				spi-max-frequency = <5000000>;   /* <= 5 MHz */
+				vref-supply = <&ain_vref>;
+				status = "okay";
+			};
+
+			mbus: rf@2 {
+				compatible = "spidev", "ti,cc1101";
+				pinctrl-names = "default";
+				pinctrl-0 = <&mbus_pins>;
+				reg = <2>;
+				spi-max-frequency = <5000000>; /* <= 6.5 MHz */
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <45 IRQ_TYPE_LEVEL_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@6 {
+		target = <&usb>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dwc2_usb: __overlay__ {
+			compatible = "brcm,bcm2835-usb";
+			dr_mode = "host";
+			g-np-tx-fifo-size = <32>;
+			g-rx-fifo-size = <558>;
+			g-tx-fifo-size = <512 512 512 512 512 256 256>;
+			status = "okay";
+
+			hub@1 {
+				/* SMSC LAN9514 */
+				compatible = "usb424,9514";
+				reg = <1>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				eth0: ethernet@1 {
+					compatible = "usb424,ec00";
+					reg = <1>;
+				};
+				eth1: ethernet@4 {
+					/* SMSC LAN9500, attached to KSZ8795 */
+					compatible = "usb424,9500";
+					reg = <4>;
+				};
+
+
+			};
+		};
+	};
+
+	fragment@7 {
+		target = <&mmc>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&emmc_gpio34>;
+			status = "okay";
+			bus-width = <4>;
+			non-removable;
+			max-frequency = <28000000>;
+			mmc-pwrseq = <&wl_on>;
+
+			wlan0: wifi@1 {
+			// Is this needed at all? Driver does not seem to work with devtree
```

Comment:
I depends. The MAYA module needs some information about firmware filename and position as well as some basic settings. Those settings can be given to the driver on module load by specifying a config-file as module parameter.

If we want the driver to be built-in into the kernel, then we need some devicetree entries. I don't see a possibility on how to provide a configuration file to a driver that is not loaded as module (but I think, if there is a possibility, you know that). The config file provided by nxp has this entry for the maya module (and ublox states, that the module will work better with cal_data_cfg set, which should be provided by ublox later on):

```
SD8978 = {
	cfg80211_wext=0xf
	max_vir_bss=1
	cal_data_cfg=none
	ps_mode=1
	auto_ds=1
	host_mlme=1
	fw_name=nxp/sdiouartiw416_combo_v0.bin
}
```
Information about devicetree entries for the maya driver can unfortunately only be found in the source code of the driver in the file moal_init.c at around line 1701. Until now I have nothing seen in any of the official documents. -> https://source.codeaurora.org/external/imx/mwifiex/tree/mxm_wifiex/wlan_src/mlinux/moal_init.c?h=lf-5.10.72_2.2.0#n1701

Reply:
As this is device specific we don't want to have this in the filesystem. This information belongs into the device-tree.

Reply:
As frank said it only uses that stupid config file

Reply:
The driver can handle devicetree entries, but it is only ducmented in the sourcecode. Please see the link in my comment above to the source repo of the driver.

Reply:
The OF code in that driver stinks: It searches the entire DT for a `sdxxx-wlan` node instead of looking up the corresponding fwnode of the platform device it's probing. So if there are multiple Maya modules in the system, they all use the same DT entry for configuration. This is so *not* upstreamable. :(